api_open_fga.go

```diff
@@ -936,20 +936,21 @@ func (a *OpenFgaApiService) CheckExecute(r ApiCheckRequest) (CheckResponse, *htt
 	}
 	// body params
 	requestBody = r.body
-	req, err := a.client.prepareRequest(r.ctx, path, httpMethod, requestBody, localVarHeaderParams, localVarQueryParams)
-	if err != nil {
-		return returnValue, nil, err
-	}
 
 	retryParams := a.client.cfg.RetryParams
 	for i := 0; i < retryParams.MaxRetry+1; i++ {
+		req, err := a.client.prepareRequest(r.ctx, path, httpMethod, requestBody, localVarHeaderParams, localVarQueryParams)
```

Comment:
Do you think there's a benefit to having a test to ensure this doesn't regress?

I think because there's not actual connection in httpmock we can't test it there, we could use [httptest](https://pkg.go.dev/net/http/httptest#NewServer) to spin up a server, point an OpenFGAClient at that and then on the first call [close the connection](https://stackoverflow.com/questions/72368886/is-there-a-way-to-drop-an-http-connection-in-golang-without-sending-anything-to), then the second call return a 200 with the expected response?